pkg/download/goget/goget_test.go

```diff
@@ -1,40 +1,200 @@
 package goget
 
 import (
+	"bytes"
 	"context"
+	"encoding/json"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
 	"testing"
+	"time"
 
+	"github.com/gomods/athens/pkg/storage"
 	"github.com/stretchr/testify/require"
 )
 
-type testCase struct {
-	name    string
-	mod     string
-	version string
+type listTest struct {
+	name string
+	path string
+	tags []string
 }
 
-// TODO(marwan): we should create Test Repos under github.com/gomods
-// so we can get reproducible results from live VCS repos.
-// For now, I cannot test that github.com/pkg/errors returns v0.8.0
-// from goget.Latest, because they could very well introduce a new tag
-// in the near future.
-var tt = []testCase{
-	{"basic list", "github.com/pkg/errors", "latest"},
-	{"list non tagged", "github.com/marwan-at-work/gowatch", "latest"},
-	{"list vanity", "golang.org/x/tools", "latest"},
+var listTests = []listTest{
+	{
+		name: "happy tags",
+		path: "github.com/athens-artifacts/happy-path",
+		tags: []string{"v0.0.1", "v0.0.2", "v0.0.3"},
+	},
+	{
+		name: "no tags",
+		path: "github.com/athens-artifacts/no-tags",
+	},
 }
 
 func TestList(t *testing.T) {
 	dp, err := New()
 	require.NoError(t, err, "failed to create protocol")
 	ctx := context.Background()
 
-	for _, tc := range tt {
+	for _, tc := range listTests {
+		t.Run(tc.name, func(t *testing.T) {
+			versions, err := dp.List(ctx, tc.path)
+			require.NoError(t, err)
+			require.EqualValues(t, tc.tags, versions)
```

Comment:
are we going to mimick this?

https://github.com/golang/go/blob/b7d3f4c0b2f421c6b7bf6ce3533f1b34b470b39d/src/cmd/go/internal/modload/help.go#L238-L239

Reply:
^ this is obviously for @latest endpoint

Reply:
@michalpristas yeah I actually got confused about that. From what I can tell, `@latest` is never called for an actual tag. cmd/go always calls `/@v/list` first then if the list is empty it will call `@latest`. But maybe we should just follow what the spec says? I can do that in a separate issue/PR

Reply:
yeah i was thinking about exactly the same thing. not very clear to me. 
but if we take specs as a contract we probably should stick to it.